File: src/main/java/com/chrisdempewolf/pinterest/responses/pin/Attribution.java
package com.chrisdempewolf.pinterest.responses.pin;

import com.google.gson.annotations.SerializedName;

public class Attribution {
    private final String title;
    private final String url;
    @SerializedName("provider_icon_url")
    private final String providerIconUrl;
    @SerializedName("author_name")
    private final String authorName;
    @SerializedName("provider_favicon_url")
    private final String providerFaviconUrl;
    @SerializedName("author_url")
    private final String authorUrl;
    @SerializedName("provider_name")
    private final String providerName;

    public Attribution(String title, String url, String providerIconUrl, String authorName, String providerFaviconUrl, String authorUrl, String providerName) {
        this.title = title;
        this.url = url;
        this.providerIconUrl = providerIconUrl;
        this.authorName = authorName;
        this.providerFaviconUrl = providerFaviconUrl;
        this.authorUrl = authorUrl;
        this.providerName = providerName;
    }

    public String getTitle() {
        return title;
    }

    public String getUrl() {
        return url;
    }

    public String getProviderIconUrl() {
        return providerIconUrl;
    }

    public String getAuthorName() {
        return authorName;
    }

    public String getProviderFaviconUrl() {
        return providerFaviconUrl;
    }

    public String getAuthorUrl() {
        return authorUrl;
    }

    public String getProviderName() {
        return providerName;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Attribution that = (Attribution) o;

        if (title != null ? !title.equals(that.title) : that.title != null) return false;
        if (url != null ? !url.equals(that.url) : that.url != null) return false;
        if (providerIconUrl != null ? !providerIconUrl.equals(that.providerIconUrl) : that.providerIconUrl != null)
            return false;
        if (authorName != null ? !authorName.equals(that.authorName) : that.authorName != null) return false;
        if (providerFaviconUrl != null ? !providerFaviconUrl.equals(that.providerFaviconUrl) : that.providerFaviconUrl != null)
            return false;
        if (authorUrl != null ? !authorUrl.equals(that.authorUrl) : that.authorUrl != null) return false;
        return providerName != null ? providerName.equals(that.providerName) : that.providerName == null;

    }

    @Override
    public int hashCode() {
        int result = title != null ? title.hashCode() : 0;
        result = 31 * result + (url != null ? url.hashCode() : 0);
        result = 31 * result + (providerIconUrl != null ? providerIconUrl.hashCode() : 0);
        result = 31 * result + (authorName != null ? authorName.hashCode() : 0);
        result = 31 * result + (providerFaviconUrl != null ? providerFaviconUrl.hashCode() : 0);
        result = 31 * result + (authorUrl != null ? authorUrl.hashCode() : 0);
        result = 31 * result + (providerName != null ? providerName.hashCode() : 0);
        return result;
    }

    @Override
    public String toString() {
        return "Attribution{" +
                "title='" + title + '\'' +
                ", url='" + url + '\'' +
                ", providerIconUrl='" + providerIconUrl + '\'' +
                ", authorName='" + authorName + '\'' +
                ", providerFaviconUrl='" + providerFaviconUrl + '\'' +
                ", authorUrl='" + authorUrl + '\'' +
                ", providerName='" + providerName + '\'' +
                '}';
    }
}

File: src/main/java/com/chrisdempewolf/pinterest/responses/pin/Attribution.kt
package com.chrisdempewolf.pinterest.responses.pin

import com.google.gson.annotations.SerializedName

class Attribution(val title: String?, val url: String?, @SerializedName("provider_icon_url")
val providerIconUrl: String?, @SerializedName("author_name")
                  val authorName: String?, @SerializedName("provider_favicon_url")
                  val providerFaviconUrl: String?, @SerializedName("author_url")
                  val authorUrl: String?, @SerializedName("provider_name")
                  val providerName: String?) {

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val that = o as Attribution?

        if (if (title != null) title != that!!.title else that!!.title != null) return false
        if (if (url != null) url != that.url else that.url != null) return false
        if (if (providerIconUrl != null) providerIconUrl != that.providerIconUrl else that.providerIconUrl != null)
            return false
        if (if (authorName != null) authorName != that.authorName else that.authorName != null) return false
        if (if (providerFaviconUrl != null) providerFaviconUrl != that.providerFaviconUrl else that.providerFaviconUrl != null)
            return false
        if (if (authorUrl != null) authorUrl != that.authorUrl else that.authorUrl != null) return false
        return if (providerName != null) providerName == that.providerName else that.providerName == null

    }

    override fun hashCode(): Int {
        var result = if (title != null) title.hashCode() else 0
        result = 31 * result + if (url != null) url.hashCode() else 0
        result = 31 * result + if (providerIconUrl != null) providerIconUrl.hashCode() else 0
        result = 31 * result + if (authorName != null) authorName.hashCode() else 0
        result = 31 * result + if (providerFaviconUrl != null) providerFaviconUrl.hashCode() else 0
        result = 31 * result + if (authorUrl != null) authorUrl.hashCode() else 0
        result = 31 * result + if (providerName != null) providerName.hashCode() else 0
        return result
    }

    override fun toString(): String {
        return "Attribution{" +
                "title='" + title + '\'' +
                ", url='" + url + '\'' +
                ", providerIconUrl='" + providerIconUrl + '\'' +
                ", authorName='" + authorName + '\'' +
                ", providerFaviconUrl='" + providerFaviconUrl + '\'' +
                ", authorUrl='" + authorUrl + '\'' +
                ", providerName='" + providerName + '\'' +
                '}'
    }
}
